core/src/main/java/com/linecorp/armeria/client/HttpRequestSubscriber.java

```diff
@@ -218,7 +242,12 @@ public void onNext(HttpObject o) {
     @Override
     public void onError(Throwable cause) {
         isSubscriptionCompleted = true;
-        failAndRespond(cause);
+        if (id >= 0) { // onSubscribe is called.
+            failAndReset(cause);
+        } else {
+            // No need to send RST because we didn't send any packet.
+            fail(new UnprocessedRequestException(cause));
```

Comment:
Question: We don't need to care about the type of cause just in case the cause is instance of `UnprocessedRequestException` already? 
If the cause can not be `UnprocessedRequestException`, skip this review 😀

Reply:
I think that's not going to happen. If so, I should unwrap the cause and call `failAndReset` on line 246. I think it's not worth. 😄

Reply:
Thanks for the explanation! 🙏